internal/server/admin/key_test.go

```diff
@@ -0,0 +1,285 @@
+package admin
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/gavv/httpexpect/v2"
+	"github.com/google/uuid"
+	v1 "github.com/kong/koko/internal/gen/grpc/kong/admin/model/v1"
+	"github.com/stretchr/testify/require"
+)
+
+func goodKey() *v1.Key {
+	return &v1.Key{
+		Id: uuid.NewString(),
+		//nolint: lll // JSON strings can't be wrapped
+		Jwk: `{
+			"kty": "RSA",
+			"kid": "vsR8NCNV_1_LB06LqudGa2r-T0y4Z6VQVYue9IQz6A4",
+			"n": "v2KAzzfruqctVHaE9WSCWIg1xAhMwxTIK-i56WNqPtpWBo9AqxcVea8NyVctEjUNq_mix5CklNy3ru7ARh7rBG_LU65fzs4fY_uYalul3QZSnr61Gj-cTUB3Gy4PhA63yXCbYRR3gDy6WR_wfis1MS61j0R_AjgXuVufmmC0F7R9qSWfR8ft0CbQgemEHY3ddKeW7T7fKv1jnRwYAkl5B_xtvxRFIYT-uR9NNftixNpUIW7q8qvOH7D9icXOg4_wIVxTRe5QiRYwEFoUbV1V9bFtu5FLal0vZnLaWwg5tA6enhzBpxJNdrS0v1RcPpyeNP-9r3cUDGmeftwz9v95UQ",
+			"e": "AQAB",
+			"alg": "A256GCM"
+		}`,
+		Name: "simpleKey-" + uuid.NewString(),
+	}
+}
+
+func validateGoodKey(body *httpexpect.Object) {
+	body.ContainsKey("id")
+	body.ContainsKey("created_at")
+	body.ContainsKey("updated_at")
+	body.ContainsKey("jwk")
+	body.ContainsKey("kid")
+}
+
+func TestKeyCreate(t *testing.T) {
```

Comment:
Can you please add a test creating a Key which references a KeySet?

Reply:
those are in the `TestKeyAndKeyset` function

Reply:
Ops, you're right